core/whistle-1.0.0/src/wh_call_response.erl

```diff
@@ -116,18 +122,19 @@ do_send(CallId, CtrlQ, Commands) ->
 %%
 %% @end
 %%--------------------------------------------------------------------
--spec send_default(ne_binary(), ne_binary(), api_binary()) ->
+-spec send_default(whapps_call:call(), api_binary()) ->
                           {'ok', ne_binary()} |
                           {'error', 'no_response'}.
-send_default(_, _, 'undefined') ->
+send_default(_Call, 'undefined') ->
     {'error', 'no_response'};
-send_default(CallId, CtrlQ, Cause) ->
+send_default(Call, Cause) ->
     lager:debug("attempting to send default response for ~s", [Cause]),
-    Response = get_response(Cause),
-    send(CallId, CtrlQ
-         ,wh_json:get_value(<<"Code">>, Response)
-         ,wh_json:get_value(<<"Message">>, Response)
-         ,wh_json:get_value(<<"Media">>, Response)).
+    Response = get_response(Cause, Call),
+    send(cf_exe:callid(Call)
+        ,cf_exe:control_queue(Call)
```

Comment:
cf_exe belongs to callflow. this way we are making core dependent of callflow.
can you use whapps_call  ?